content/blog/2022/2022-01-18-getting-started-with-spring-boot.md

```diff
@@ -0,0 +1,915 @@
+
+We have three tables in the database:
+- `user`
+- `book`
+- `borrowed_books`
+
+The `user` table contains the id, name, last name, email, and password. The id is our primary key, and the database will autogenerate it. We will show how to do it in the [next chapter](#creating-the-entity).
+
+The `book` table has the id, title, author, publication, and number of instances in the bookstore.
+
+The `borrowed_books` table represents the `many-to-many` relationship. The `many-to-many` relationship means that one user can borrow several books and that one book can be borrowed several times.
+
+We can read more about relationships between entities in the [Handling Associations Between Entities with Spring Data REST](https://reflectoring.io/relations-with-spring-data-rest/) article.
+### Creating the Entity
+After we define the database, we can start implementing entity classes.
+The entity class is the direct connection to the table in the database:
+```java
+@Entity(name = "book")
+public class Book {
+
+    //The rest of the class omitted
+
+}
+```
+The `@Entity` annotation indicates that the annotated class is a JPA entity. The `name` attribute inside the annotation indicates the table name. 
+Defining the name attribute is not mandatory, but if we do not set the `name` attribute, Spring will assume that the table name is the same as the class name.
+
+### Defining a Primary Key
+```java
+@Entity(name = "book")
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+    //The rest of the class omitted
+
+}
+```
+When defining a class as an entity, we need to provide the id column. The id column will be the primary key of that table.
+
+We need to decide how will the id column be generated. Using the `@GeneratedValue` annotation we can define several different strategies:
+- [IDENTITY](#identity-generation-strategy)
+- [SEQUENCE](#sequence-generation-strategy)
+- [TABLE](#table-generation-strategy)
+- [AUTO](#defining-the-primary-keys)
+
+We used the `AUTO` strategy, which leverages whatever the database prefers. Most databases prefer to use the `SEQUENCE` strategy for their primary key definition.
+
+#### Identity Generation Strategy
+The `GenerationType.IDENTITY` strategy allows the database to autoincrement the id value when we insert a new row.
+Let us look into the example of using the identity generation strategy:
+```java
+@Entity(name = "book")
+public class Book {
+
+    @Id
+    @GeneratedValue( strategy = GenerationType.IDENTITY)
+    private long id;
+    
+    // Rest of the code omitted
+}
+```
+We define the identity generation type in the strategy attribute of the `@GeneratedValude`. While the identity strategy is highly efficient for the database, it doesn't perform well with the Hibernate ORM.
+The Hibernate expects that every managed entity has its id set, so it needs to go and call the database to insert the id.
```

Comment:
What is Hibernate? This is the first time it's mentioned and as a newbie I'm confused :). Maybe mention it in the section about the JPA dependency above. Also explain what JPA stands for, while you're at it.